src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -139,11 +128,24 @@ public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(
      * @return a textSubmission without any manual result or an empty Optional if no submission without manual result could be found
      */
     public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(TextExercise textExercise, boolean skipAssessmentQueue, boolean examMode, int correctionRound) {
+        var assessableSubmissions = getAssessableSubmissions(textExercise, examMode, correctionRound);
+
         // If automatic assessment is enabled and available, try to learn the most possible amount during the first correction round
-        if (textExercise.isAutomaticAssessmentEnabled() && textAssessmentQueueService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
-            return textAssessmentQueueService.get().getProposedTextSubmission(textExercise);
+        if (textExercise.isFeedbackSuggestionsEnabled() && athenaSubmissionSelectionService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
+            var athenaSubmissionId = athenaSubmissionSelectionService.get().getProposedSubmissionId(textExercise, assessableSubmissions.stream().map(Submission::getId).toList());
+            if (athenaSubmissionId.isPresent()) {
+                var submission = textSubmissionRepository.findWithEagerResultsAndFeedbackAndTextBlocksById(athenaSubmissionId.get());
+                // Test again if it is still assessable (Athena might have taken some time to respond and another assessment might have started in the meantime):
+                if (submission.isPresent() && (submission.get().getLatestResult() == null || !submission.get().getLatestResult().isManual())) {
+                    return submission;
+                }
+                else {
+                    log.debug("Athena proposed submission {} is not assessable anymore", athenaSubmissionId.get());
+                }
+            }
         }
-        var submissionWithoutResult = super.getRandomAssessableSubmission(textExercise, examMode, correctionRound);
+
+        var submissionWithoutResult = super.getRandomAssessableSubmission(assessableSubmissions);
```

Comment:
I appreciate your effort in reducing database calls, however in this case we should probably reload the assessableSubmissions again to avoid randomly selecting a submission that in the meantime got locked by another tutor.

Reply:
Makes sense 👍